src/platform/apollolake/boot_ldr.x.in

```diff
@@ -7,6 +7,7 @@
  * Use spaces for formatting as cpp ignore tab sizes.
  */
 
+#include <config.h>
```

Comment:
Why do we need to do this? These linker scripts don't use any defines from config.h.

Reply:
There is differentiation in memory.h and core-isa.h that was broken in BDW. Also having this included by default allows future Kconfig options (like build a section for debugger usage etc)

Reply:
This means, that you should include config.h in memory.h and core-isa.h instead of linker scripts. I must have missed that requirements during my header refactor, but I've cleaned up the whole source code to include only headers, that are required. Implicit inclusions are not a good practice.

Reply:
I'll fix it in core-isa.h and memory.h (since it's missing in some), but I do want to keep it here too for optional sections (including the vthread area).